lib/puppet/pops/loaders.rb

```diff
@@ -92,6 +92,10 @@ def puppet_system_loader
     @puppet_system_loader
   end
 
+  def resource_type_loader
+    @resource_type_loader
+  end
+
```

Comment:
This looks wrong conceptually as the loaders are built in a hierarchy, all things loaded fits into that hierarchy and "resource types in the environment" is not a separate hierarchical layer.
I understand that since it is loading using a 3.x loader this loader is also a module loader, and it is thus quite different from both the environment and module loaders. I think I would be happier if it was named "3xLegacyLoader" or something like that - that shows the intent in a much better way. (I think that is all that is needed to address my concern/problem both here, and where I questioned why the resource_loader was directly requested from pops bridge.